### PR TITLE
Vmware nic

### PR DESCRIPTION
**Pull request by juliedavila**
Support for creating/updating/removing NICs

### CODE REVIEW

**Comment by sivel** [comment]
At the moment, this is failing linting checks:

```
============================================================================
./cloud/vmware/vmware_nic.py
============================================================================
ERROR: indentation contains tabs. line 393 column 12
ERROR: indentation contains tabs. line 395 column 0
ERROR: indentation contains tabs. line 396 column 0
ERROR: indentation contains tabs. line 397 column 5
ERROR: indentation contains tabs. line 398 column 0
ERROR: indentation contains tabs. line 418 column 8
ERROR: indentation contains tabs. line 419 column 8
ERROR: indentation contains tabs. line 420 column 8
ERROR: indentation contains tabs. line 421 column 8
ERROR: indentation contains tabs. line 422 column 8
ERROR: indentation contains tabs. line 423 column 8
ERROR: indentation contains tabs. line 424 column 8
ERROR: indentation contains tabs. line 441 column 0
ERROR: indentation contains tabs. line 445 column 8
ERROR: Interpreter line is not "#!/usr/bin/python"
```

**Comment by juliedavila** [comment]
@sivel updated

File: cloud/vmware/vmware_nic.py [hunk 1]
```diff
@@ -0,0 +1,468 @@
...
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with Ansible.  If not, see <http://www.gnu.org/licenses/>.
+
+DOCUMENTATION = '''
+---
+module: vmware_nic
+version_added: 2.1
```
Comment (resmo) [inline cloud/vmware/vmware_nic.py]: Please use quotes for the version e.g. `version_added: "2.1"`

File: cloud/vmware/vmware_nic.py [hunk 2]
```diff
@@ -0,0 +1,468 @@
...
+options:
+  vm_path:
+    description:
+      - The fully qualified path of the VM such as /SomeFolder/SubFolder/VMname
+    required: true
+  label:
+    description:
+      - The label of the NIC you want to update or remove. This has no effect when trying to
+        create a NIC due to a limitation in VMWare's API 
+    required: false
```
Comment (resmo) [inline cloud/vmware/vmware_nic.py:34]: add a default if `required: false` even for none/null e.g. `default: none`

File: cloud/vmware/vmware_nic.py [hunk 3]
```diff
@@ -0,0 +1,468 @@
...
+  dvs:
+    description:
+      - Whether the NICs network should be standard or DVS. 
+    default: false
```
Comment (resmo) [inline cloud/vmware/vmware_nic.py]: always add `required: <true/false>`

File: cloud/vmware/vmware_nic.py [hunk 4]
```diff
@@ -0,0 +1,468 @@
...
+    description:
+      - The type of NIC you wish to create 
+    required: false
+    default: vmxnet3 
+    choices: ['vmxnet','vmxnet2','vmxnet3','e1000','e1000e','pcnet32']
+  network_name:
+    description:
+      - The name label of the network you wish to associate the NIC with. 
+        Valid for create and update states.
+    required: False
```
Comment (resmo) [inline cloud/vmware/vmware_nic.py:61]: add a default if `required: false` even for none/null e.g. `default: none`

**Comment by resmo** [comment]
Please avoid merge commits in your commits. use rebase instead.

**Comment by juliedavila** [comment]
Updated.

File: cloud/vmware/vmware_nic.py [hunk 5]
```diff
@@ -0,0 +1,470 @@
...
+     - Allows non-idempotent creation of network adapters. Updates and removals are idempotent.
+options:
+  vm_path:
+    description:
+      - The fully qualified path of the VM such as /SomeFolder/SubFolder/VMname
+    required: true
+  label:
+    description:
+      - The label of the NIC you want to update or remove.
+      - This has no effect when trying to create a NIC due to a limitation in VMWare's API
```
Comment (gundalow) [inline cloud/vmware/vmware_nic.py]: Minor thing, but its "VMware", rather than "VMWare"

File: cloud/vmware/vmware_nic.py [hunk 6]
```diff
@@ -0,0 +1,470 @@
...
+    description:
+      - Whether the NICs network should be standard or DVS.
+    required: false
+    default: false
+  state:
+    description:
+      - The action desired. If 'create', a new NIC will be created everytime.
+      - Update will check if a NIC with the same label exists, if it does then
+        it will verify that the current network matches the settings supplied in the task.
+    required: True
```
Comment (gundalow) [inline cloud/vmware/vmware_nic.py:46]: `choices: ['create', 'absent', 'update']`

**Comment by gregdek** [comment]
@defionscode A friendly reminder: this pull request has been marked as needing your action. If you still believe that this PR applies, and you intend to address the issues with this PR, just let us know in the PR itself and we will keep it open pending your changes. When you do address the issues, please respond with ready_for_review in your comment, so that we can notify the maintainer.

[This message brought to you by your friendly Ansibull-bot.]

**Comment by juliedavila** [comment]
updated

**Comment by gregdek** [comment]
Thanks @defionscode for this PR. This PR requires revisions, either because it fails to build or by reviewer request. Please make the suggested revisions. When you are done, please comment with text 'ready_for_review' and we will put this PR back into review.

[This message brought to you by your friendly Ansibull-bot.]

**Comment by gregdek** [comment]
@defionscode A friendly reminder: this pull request has been marked as needing your action. If you still believe that this PR applies, and you intend to address the issues with this PR, just let us know in the PR itself and we will keep it open pending your changes. When you do address the issues, please respond with ready_for_review in your comment, so that we can notify the maintainer.

[This message brought to you by your friendly Ansibull-bot.]

**Comment by juliedavila** [comment]
ready_for_review

**Comment by gregdek** [comment]
Thanks @defionscode. To the current maintainers, @jcpowermac, @mtnbikenc, @chrrrles please review according to guidelines (http://docs.ansible.com/ansible/developing_modules.html#module-checklist) and comment with text 'shipit', 'needs_revision' or 'close_me' as appropriate.

[This message brought to you by your friendly Ansibull-bot.]

File: cloud/vmware/vmware_nic.py [hunk 7]
```diff
@@ -0,0 +1,474 @@
...
+    required: False
+    default: null
+  datacenter:
+    description:
+      - The datacenter name in which the VM resides
+    required: True
+notes:
+  - This module should run from a system that can access vSphere directly.
+    Either by using local_action, or using delegate_to.
+author: "Jonathan Davila <jdavila@ansible.com>"
```
Comment (gundalow) [inline cloud/vmware/vmware_nic.py]: Hi, can you please change this to
`First Last (@GitHub)`

**Comment by gundalow** [comment]
Hi,

Looks like this branch needs updating to fix the Travis issue, please see this guide on how to do that cleanly.

http://docs.ansible.com/ansible/community.html#contributing-code-features-or-bugfixes

needs_revision

**Comment by gregdek** [comment]
Thanks @defionscode for this PR. This PR requires revisions, either because it fails to build or by reviewer request. Please make the suggested revisions. When you are done, please comment with text 'ready_for_review' and we will put this PR back into review.

[This message brought to you by your friendly Ansibull-bot.]

**Comment by juliedavila** [comment]
@gundalow the travis issue seems to be related to yum which this PR has nothing to do with, so not sure how to address

**Comment by gundalow** [comment]
To fix the YUM issue please rebase as shown in this document:
http://docs.ansible.com/ansible/community.html#contributing-code-features-or-bugfixes

**Comment by gregdek** [comment]
Thanks @defionscode. To the current maintainers, @jcpowermac, @mtnbikenc, @chrrrles please review according to guidelines (http://docs.ansible.com/ansible/developing_modules.html#module-checklist) and comment with text 'shipit', 'needs_revision' or 'close_me' as appropriate.

[This message brought to you by your friendly Ansibull-bot.]

**Comment by gregdek** [comment]
Thanks @defionscode for this PR. This module is maintained by the Ansible core team, so it can take a while for patches to be reviewed. Thanks for your patience.

Core team: please review according to guidelines (http://docs.ansible.com/ansible/developing_modules.html#module-checklist) and comment with 'needs_revision' or merge as appropriate.

[This message brought to you by your friendly Ansibull-bot.]

**Pull request closed without merging**